dtfs-central-api/src/v1/routes/middleware/payload-validation/validate-post-upload-utilisation-report-payload.test.ts

```diff
@@ -0,0 +1,128 @@
+      },
+    };
+
+    const getHttpMocks = (options?: Partial<Unknown<PostUploadUtilisationReportRequestBody>>) =>
+      httpMocks.createMocks({
+        body: {
+          ...aValidPayload,
+          ...options,
+        },
+      });
+
+    it('calls the next function when there are no validation errors', () => {
+      // Arrange
+      const { req, res } = getHttpMocks({
+        reportId: 1,
+        fileInfo: MOCK_AZURE_FILE_INFO,
+        reportData: [{ 'a header': 'a value', 'another header': null }],
+        user: {
+          _id: userId,
+        },
+      });
+      jest.mocked(validateUtilisationReportCsvData).mockReturnValue([]);
+
+      // Act
+      validatePostUploadUtilisationReportPayload(req, res, mockNext);
+
+      // Assert
+      expect(mockNext).toHaveBeenCalled();
+      expect(res._isEndCalled()).toBe(false);
+    });
+
+    const propertyNamesWithInvalidValues = [
+      {
+        propertyName: 'reportId',
+        value: '20',
+      },
+      {
+        propertyName: 'fileInfo',
+        value: {},
+      },
+      {
+        propertyName: 'user',
+        value: 'abc123',
+      },
+    ] as const;
+
+    it.each(propertyNamesWithInvalidValues)("responds with an error if the '$propertyName' property is invalid", ({ propertyName, value }) => {
+      // Arrange
+      const { req, res } = getHttpMocks({
+        [propertyName]: value,
+      });
+
+      // Act
+      validatePostUploadUtilisationReportPayload(req, res, mockNext);
+
+      // Assert
+      expect(mockNext).not.toHaveBeenCalled();
+      expect(res._getStatusCode()).toBe(HttpStatusCode.BadRequest);
+    });
+
+    it.each`
+      condition                        | value
+      ${'undefined'}                   | ${undefined}
+      ${'null'}                        | ${null}
+      ${'object'}                      | ${{}}
+      ${'string'}                      | ${'some data'}
+      ${'item properties not strings'} | ${[{ key: {} }]}
+    `('responds with an error if the reportData is invalid (case: $condition)', ({ value }: { value: unknown }) => {
+      // Arrange
+      const { req, res } = getHttpMocks({
+        reportData: value,
+      });
+
+      // Act
+      validatePostUploadUtilisationReportPayload(req, res, mockNext);
+
+      // Assert
+      expect(mockNext).not.toHaveBeenCalled();
+      expect(res._getStatusCode()).toBe(HttpStatusCode.BadRequest);
+    });
+
+    it('responds with an error if the reportData contains validation errors', () => {
+      // Arrange
+      const { req, res } = getHttpMocks({
+        reportData: [{ 'a header': 'a value' }],
+      });
+      jest.mocked(validateUtilisationReportCsvData).mockReturnValue([{ errorMessage: 'That is not valid report data' }]);
+
+      // Act
+      validatePostUploadUtilisationReportPayload(req, res, mockNext);
+
+      // Assert
+      expect(mockNext).not.toHaveBeenCalled();
+      expect(res._getStatusCode()).toBe(HttpStatusCode.BadRequest);
+    });
```

Comment:
Can we add a test to check the validation errors are returned in the response? e.g

```ts
expect(res.send).toHaveBeenCalledWith(expected);
```